smartsim/experiment.py

```diff
@@ -168,11 +168,21 @@ def __init__(
         if self._launcher == "cobalt":
             raise SSUnsupportedError("Cobalt launcher is no longer supported.")
 
+        if launcher == "dragon":
```

Comment:
Could we add something to a launcher that does prep work and avoid coupling the experiment?

```
DragonLauncher(WLMLauncher):
  def init_environment(self) -> None:
    self._set_dragon_server_path()
```

THis feels like it might be circular....

Maybe a better approach would be to just inject callables into the `Experiment` or import and alias a callable...

Maybe not worth the complexity at this time. Feel free to ignore.

Reply:
It _would be worth the complexity_ but we need information that only the `Experiment` has and would have to be set to the launcher attribute of the controller, which means that the Experiment should be available there. We currently do not have a back reference (although we may want to add one in the future)